lib/TableGen/Operations.cpp

```diff
@@ -459,6 +459,11 @@ void BuilderMethod::emitDefinition(raw_ostream &out, FmtContext &fmt,
   fmt.addSubst("fn", symbols.chooseName({"fn", "function"}));
   out << tgfmt(R"(
     auto $fn = $_module.getOrInsertFunction($fnName, $fnType, $attrs);
+    for (unsigned i = 0; ::llvm::isa<::llvm::ConstantExpr>($fn.getCallee()); i++) {
+      // Try new names until we get one with the correct type
+      std::string newName = (::llvm::Twine($fnName) + "_" + ::llvm::utostr(i)).str();
```

Comment:
Not sure whether it's worth it, but we could use a `SmallString` with a `llvm::raw_svector_ostream` instead.